frontend_tests/node_tests/activity.js

```diff
@@ -92,15 +102,47 @@ activity.update_huddles = function () {};
 global.compile_template('user_presence_row');
 global.compile_template('user_presence_rows');
 
+var presence_info = {};
+presence_info[alice.user_id] = { status: 'inactive' };
+presence_info[fred.user_id] = { status: 'active' };
+presence_info[jill.user_id] = { status: 'active' };
+
+presence.presence_info = presence_info;
+
+(function test_get_status() {
+    assert.equal(presence.get_status(page_params.user_id), "active");
+    assert.equal(presence.get_status(alice.user_id), "inactive");
+    assert.equal(presence.get_status(fred.user_id), "active");
+}());
+
+(function test_last_seen_status_from_date() {
+    var last_seen_base_date = new XDate(2016, 2, 29, 10, 30);
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addSeconds(-20), last_seen_base_date), "Last seen just now");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addMinutes(-1), last_seen_base_date), "Last seen a minute ago");
+    assert.equal(timerender.last_seen_status_from_date(last_seen_base_date.clone().addMinutes(-2), last_seen_base_date), "Last seen 2 minutes ago");
```

Comment:
I think you can make these tests more readable with a helper function:

~~~ js
    var base_date = new XDate(2016, 2, 29, 10, 30);

    function assert_same (modifier, expected_string) {
        var past_date = base_date.clone();
        past_date = modifier(past_date);
        var actual_string = timerender.last_seen_status_from_date(past_date, base_date);
        assert.equal(actual_string, expected_string);
    }   

    assert_same(function (d) { return d.addSeconds(-20); },
                "Last seen just now");

    assert_same(function (d) { return d.addMinutes(-1); },
                "Last seen one minute ago");

    assert_same(function (d) { return d.addMinutes(-2); },
                "Last seen two minutes ago");